scrapers/timeanddate.js

```diff
@@ -1,63 +1,61 @@
-/* Usage : timeanddate.js London
-           timeanddate.js 
- */
 
-const request = require('request-promise-native');
-const cheerio = require('cheerio');
+var BaseLoklakScrapper = require('./base');
 
+class LocationAndDateScrapper extends BaseLoklakScrapper {
 
-var url;
-var html;
-var $;
+  constructor() {
+    super('LocationAndDate', 'http://www.timeanddate.com/worldclock/results.html?query=');
+  }
 
-if (process.argv.length === 3) {
-  query = process.argv[2];
-  terms = ['all'];
-}
+  argumentSanityCheck(args) {
+    super.argumentSanityCheck(args);
 
-url = "http://www.timeanddate.com/worldclock/results.html?query=" + query;
+	if (args.length <= 2) {
+      console.error('Atleast one argument required.');
+      process.exit(-1);
+    }
 
-request(url, function(error, response, body) {
-  if(error) {
-    console.log("Error: " + error);
-    process.exit(-1);
+    return true;
   }
-  html = body;
-  scrapeTimeAndDate()
-});
-
-
-function scrapeTimeAndDate() {
-  $ = cheerio.load(html);
-  var loc_list = {};
-  var htmlTime = $("table");
-  var tag, location, time;
-  var locationwisetime = [];
-  var count = 0;
-
-  $('table').find('tr').each(function (index, element) {
-
-  tag = $(element).find("td");
-  if( tag.text() != "") {
-    location = tag.text();
-    tag = tag.next();
-    time = tag.text();
-    location = location.replace(time,"");
-
-    loc_list["location"] = location;
-    loc_list["time"] = time;
-    locationwisetime.push(loc_list);
-    count++;
-    loc_list = {};
-  } else {
-  tag = tag.next();
+
+  onInit() {
+    this.REQUEST_URL = this.BASE_URL + this.SLICED_PROC_ARGS[0];
+    this.request();
   }
-    
 
-});
-  loc_list["count"] = count;
-  locationwisetime.push(loc_list);
-  console.log(locationwisetime);
+  scrape($) {
+    const locationwisetime = [];
+    var loc_list = {};
```

Comment:
use `let` if the variables are modified else use `const`. using `var` in ES6 is an `anti-practice`